qcodes/instrument/parameter.py

```diff
@@ -969,6 +969,18 @@ def settable(self) -> bool:
         """
         return self._settable
 
+    @property
+    def underlying_instrument(self) -> Optional['InstrumentBase']:
+        """
+        Property to expose virtual parameters' underlying instrument i.e.
+        to allow the user to set the actual instrument of a parameter to be
+        the underlying instrument for a virtual parameter. A virtual
+        parameter is one which is built over another existing parameter.
+
+        By default it return the root_instrument of the parameter.
```

Comment:
@Akshita07 i was trying to improve this docstring for clarity, and this is what i came up with, what do you think? is this better?
```suggestion
        Returns an instance of the underlying hardware instrument that this
        parameter communicates with, per this parameter's implementation.

        This is useful in the case where a parameter does not belongs to
        an instrument instance that represents a real hardware instrument
        but actually uses a real hardware instrument in its implementation
        (e.g. via calls to one or more parameters of that real hardware
        instrument). This is also useful when a parameter does belong to
        an instrument instance but that instance does not represent the
        real hardware instrument that the parameter interacts with: hence
        ``root_instrument`` of the parameter cannot be the
        ``hardware_instrument``, however ``underlying_instrument`` can be
        implemented to return the ``hardware_instrument``.

        By default it returns the ``root_instrument`` of the parameter.
```

Reply:
@astafan8 this reads much better :). I am comiting it and resolving this comment.